engine/src/main/java/org/terasology/config/flexible/FlexibleConfig.java

```diff
@@ -54,4 +57,10 @@
      * Returns a boolean stating whether the config contains a {@link Setting} with the given id.
      */
     boolean contains(SimpleUri id);
+
+    void setUnusedSettings(Map<SimpleUri, String> unusedSettings);
+
+    Map<SimpleUri, String> getUnusedSettings();
+
+    Map<SimpleUri, Setting> getActiveSettings();
```

Comment:
I don't understand the difference between an unused and an active setting from the api, maybe add some javadoc to explain the concept?

Reply:
I was planning on doing that after the code stabilizes a bit, but I'll explain some comments explaining the basic logic for reviewers' benifit.